### PR TITLE
UBERF-4472-sharing-views

### PR DESCRIPTION
**Pull request by Syarg**
# Contribution checklist

## Brief description
Added copy link functionality to saved views
image
Copying to buffer link like
http://localhost:8080/workbench/ws1/recruit/vacancies?filterVeiewId=658a7231db7437e749fccc49
Following that link opens corresponding view if it's available to the user using link.

image
image
Added public/non public switch to saved view popup menu.

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does a local formatting is applied (rush format)
* [ ] - Does a local svele-check is applied (rush svelte check)
* [ ] - Does a local UI tests are executed [UI Testing](../tests/readme.md)
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

## Contributor requirements

* [ ] - Sign-off is provided. [DCO](https://github.com/apps/dco)
* [ ] - GPG Signature is provided. [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

### CODE REVIEW

File: plugins/workbench-resources/src/components/SavedView.svelte [hunk 1]
```diff
@@ -49,6 +52,14 @@
     (result) => {
       myFilteredViews = result.filter((p) => p.users.includes(me))
       availableFilteredViews = result.filter((p) => p.sharable && !p.users.includes(me))
+
+      const location = getLocation()
+      if (location.query?.filterVeiewId) {
```
Comment (aonnikov) [inline plugins/workbench-resources/src/components/SavedView.svelte]: typo: `filterVeiewId` -> `filterViewId`

File: plugins/view-resources/src/components/navigator/TreeElement.svelte [hunk 2]
```diff
@@ -47,13 +48,25 @@
   export let actions: (originalEvent?: MouseEvent) => Promise<Action[]> = async () => []
 
   let hovered = false
+  let inlineActions: Action[] = []
+  let popupMenuActions: Action[] = []
+
+  $: actions().then((result) => {
+    inlineActions = result.filter((action) => action.inline === true)
+    popupMenuActions = result.filter((action) => !action.inline === true)
```
Comment (aonnikov) [inline plugins/view-resources/src/components/navigator/TreeElement.svelte]: `!action.inline === true` -> `action.inline === false` or `action.inline !== true` if it is nullable

File: plugins/workbench-resources/src/components/SavedView.svelte [hunk 3]
```diff
@@ -85,13 +96,52 @@
     ]
   }
 
+  async function switchPublicAction (object: FilteredView, originalEvent: MouseEvent | undefined): Promise<Action[]> {
+    return [
+      {
+        icon: object.sharable ? task.icon.TodoCheck : task.icon.TodoUnCheck,
+        label: view.string.PublicView,
+        action: async (ctx: any, evt: Event) => {
+          await client.update(object, { sharable: !object.sharable })
```
Comment (aonnikov) [inline plugins/workbench-resources/src/components/SavedView.svelte:105]: What happens if I unshare shared view that someone already uses?

File: plugins/view-resources/src/components/navigator/TreeElement.svelte [hunk 4]
```diff
@@ -91,6 +104,19 @@
   <div class="an-element__grow" />
 
   {#if !parent}
+    {#each inlineActions as action}
+      <div
+        class="an-element__tool"
+        class:pressed={hovered}
+        on:click|preventDefault|stopPropagation={(ev) => onInlineClick(ev, action)}
+      >
+        {#if action.icon !== undefined}
```
Comment (aonnikov) [inline plugins/view-resources/src/components/navigator/TreeElement.svelte]: nit: can be simplified to one line:
`<Icon icon={action.icon ?? ActionIcon} size={'small'} />`

File: plugins/workbench-resources/src/components/SavedView.svelte (hunk 3 above)
Comment (Syarg) [reply plugins/workbench-resources/src/components/SavedView.svelte:105]: Hm, interesting one, need to check on already saved ones.

Comment (Syarg) [reply plugins/workbench-resources/src/components/SavedView.svelte:105]: View made non-public is left as is for all users already added it, but becomes not visible in view search for adding.

**Review by haiodo — APPROVED** [review]

**Pull request merged**